Flaky-test: BrokerRegistryTest.testRegisterAndLookup

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/4073474341/jobs/7019289431#step:11:1397

### Exception stacktrace

```
  Error:  Tests run: 12, Failures: 1, Errors: 0, Skipped: 8, Time elapsed: 18.168 s <<< FAILURE! - in org.apache.pulsar.broker.loadbalance.extensions.BrokerRegistryTest
  Error:  testRegisterAndLookup(org.apache.pulsar.broker.loadbalance.extensions.BrokerRegistryTest)  Time elapsed: 5.181 s  <<< FAILURE!
  java.lang.AssertionError: expected [2] but found [3]
  	at org.testng.Assert.fail(Assert.java:110)
  	at org.testng.Assert.failNotEquals(Assert.java:1413)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
  	at org.testng.Assert.assertEquals(Assert.java:131)
  	at org.testng.Assert.assertEquals(Assert.java:1240)
  	at org.testng.Assert.assertEquals(Assert.java:1274)
  	at org.apache.pulsar.broker.loadbalance.extensions.BrokerRegistryTest.testRegisterAndLookup(BrokerRegistryTest.java:277)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

Logs: https://gist.github.com/lhotari/4e4a840de15e19774213fe6d034db3f2

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/extensions/BrokerRegistryTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/extensions/BrokerRegistryTest.java
@@ -211,8 +211,10 @@ public class BrokerRegistryTest {
         brokerRegistries.clear();
         log.info("Cleaning up the pulsar services...");
         pulsarServices.forEach(pulsarService -> {
-            if (pulsarService.isRunning()) {
-                pulsarService.shutdownNow();
+            try {
+                pulsarService.close();
+            } catch (PulsarServerException e) {
+                throw new RuntimeException(e);
             }
         });
         pulsarServices.clear();
@@ -274,8 +276,12 @@ public class BrokerRegistryTest {
 
         // Unregister and see the available brokers.
         brokerRegistry1.unregister();
-        assertEquals(brokerRegistry2.getAvailableBrokersAsync().get().size(), 2);
 
+        // After unregistering, the other broker registry lock manager's metadata cache might not be updated yet,
+        // so we need to wait for it to synchronize.
+        Awaitility.await().untilAsserted(() -> {
+            assertEquals(brokerRegistry2.getAvailableBrokersAsync().get().size(), 2);
+        });
     }
 
     @Test

